src/wrapper/simulation_objects.py

```diff
@@ -216,37 +324,154 @@ def add_subscriptions(self) -> int:
+        """
+        return (self._edges[0].from_node, self._edges[0].to_node)
+
+    def __init__(self, edge1, edge2):
+        if edge1.identifier.endswith("-re"):
+            assert edge1.identifier.split("-re")[0] == edge2.identifier
+            self._edges = (edge1, edge2)
+        else:
+            assert edge1.identifier == edge2.identifier.split("-re")[0]
+            self._edges = (edge2, edge1)
+
+        super().__init__(identifier=self._edges[0].identifier)
+
+        edge1.track = self
+        edge2.track = self
+
+    @property
+    def max_speed(self) -> Tuple[float, float] | float:
+        """Returns the max_speed of the track
+
+        :return: If both edges have the same speed, a float is returned,
+        if the edges have different speeds, a tuple is returned
+        """
+        speed0 = self.edges[0].max_speed
+        speed1 = self.edges[1].max_speed
+
+        if speed0 == speed1:
+            return speed0
+
+        return (speed0, speed1)
+
+    @max_speed.setter
+    def max_speed(self, speed: Tuple[float, float] | float) -> None:
+        """Updates the max_speed of the track
+
+        :param speed: If both edges should have the same speed, a floatshould be passed,
+        if the edges should have different speeds, pass a tuple
+        """
+        if not isinstance(speed, Tuple):
+            speed = (speed, speed)
+
+        self.edges[0].max_speed = speed[0]
+        self.edges[1].max_speed = speed[1]
+
+    @property
+    def blocked(self) -> Tuple[bool, bool] | bool:
+        """Returns if the edge/track is blocked.
+
+        :return: If only one edge is blocked, a tuple is returned;
+        if both edges are blocked / unblocked, a plain bool is returned
+        """
+        if self.edges[0].blocked == self.edges[1].blocked:
+            return self.edges[0].blocked
+        return (self.edges[0].blocked, self.edges[1].blocked)
+
+    @blocked.setter
+    def blocked(self, blocked: Tuple[bool, bool] | bool) -> None:
+        """Block the edges represented by this track
+
+        :param blocked: Which edge to block (if only one direction should be blocked,
+        use the tuple, else use the plain bool)
+        """
+        if not isinstance(blocked, Tuple):
+            blocked = (blocked, blocked)
+
+        self.edges[0].blocked = blocked[0]
+        self.edges[1].blocked = blocked[1]
+
+    def update(self, data: dict) -> None:
+        # pylint: disable=unused-argument
+        pass
+
+    def add_subscriptions(self):
+        pass
+
+    @staticmethod
+    def from_simulation(
+        simulation_object, updater: "SimulationObjectUpdatingComponent"
+    ) -> "SimulationObject":
+        pass
+
+    def add_simulation_connections(self) -> None:
         pass
 
 
 class Platform(SimulationObject):
     """A platform where trains can arrive, load and unload passengers and depart"""
 
-    _track: Track = None
-    _track_id: str
+    _edge: Edge = None
+    _edge_id: str
```

Comment:
Do I understand that correctly: The two tracks for different directions are not seperate elements in the simulation network?

Reply:
I am not sure what your question is: We have the following objects available: 
- `Track`: no direction information available, composed of two edges
- `Edge`: directional, reference to the track

A `platform` has a reference to its edge, in order to mirror the behavior of SUMO.

Reply:
Okay, got it! Thanks!